packages/libs/client/src/client/utils/tests/utils.test.ts

```diff
@@ -81,70 +81,67 @@ describe('client utils', () => {
       expect(obj.reject).toBeDefined();
     });
 
-    it('resolves the promise if resolve function is called', (done) => {
+    it('resolves the promise if resolve function is called', () => {
       const pr = getPromise();
       pr.promise
         .then((result) => {
           expect(result).toBe('result');
-          done();
         })
         .catch(() => {
           expect('this should not happen').toBe(true);
-          done();
         });
       pr.resolve('result');
+      return pr;
```

Comment:
I know this isn't part of your pr, but why are we writing this test so unlike our own writing style? Wouldn't we normally write something more akin to:
```ts
it('resolves the promise if resolve function is called', async () => {
  const result = await getPromise();
  expect(result).toBe('result');
});
```

Reply:
Good point, totally agree. Didn't bother to pick that up in this PR.

Reply:
We cant do `const result = await getPromise();` this will wait forever!, you need to call the resolve function to resolve this promise. check line 93 (`pr.resolve('result');`)
so @webpro's approach is better